src/dispatcher/server.rs

```diff
@@ -53,26 +48,28 @@ impl SignatureDispatcher for Ed25519SignatureDispatcher {
         &self,
         request: Request<SignMilestoneRequest>,
     ) -> Result<Response<SignMilestoneResponse>, Status> {
-
         debug!("Got Request: {:?}", request);
 
         let r = request.get_ref();
         // Check that the pubkeys do not repeat
         let pub_keys_unique = r.pub_keys.iter().unique();
         // We do not need to check for the lexicographical sorting of the keys, it is not our job
 
-        let matched_signers = pub_keys_unique.map(|pubkey| {
-            self.keysigners.iter().find(
-                |keysigner| keysigner.pubkey == *pubkey
-            )
+        let keysigners_guard = self.keysigners.lock().await;
+        let mut matched_signers = pub_keys_unique.map(|pubkey| {
```

Comment:
The `mut` can now go.
```suggestion
        let matched_signers = pub_keys_unique.map(|pubkey| {
```